tests/shared/src/test/scala/org/http4s/headers/SourceMapSuite.scala

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2013 http4s.org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.http4s
+package headers
+
+import cats.effect.IO
+import org.http4s.syntax.header._
+
+class SourceMapSuite extends HeaderLaws {
+
+  def getUri(uri: String): Uri =
+    Uri.fromString(uri).fold(_ => sys.error(s"Failure on uri: $uri"), identity)
```

Comment:
`sys.error` looks too strict, wouldn't it be fine just to throw an exception here?

Reply:
done

Reply:
Oh snap! 😞 Didn't wake up properly and didn't suggest using URI literal syntax. Fixed on my own, hope you don't mind!